viscy/light/engine.py

```diff
@@ -745,6 +751,15 @@ def predict_step(self, batch: TripletSample, batch_idx, dataloader_idx=0):
         return features, projections, index
 
     def on_predict_epoch_end(self) -> None:
+        if (
+            not self.features_output_path
+            or not self.projections_output_path
+            or not self.metadata_output_path
+        ):
+            raise ValueError(
+                "Output paths for features, projections, and metadata must be provided."
+            )
+
```

Comment:
Should be in `on_predict_start` instead?

Reply:
fixed